max/kernels/src/layout/tensor_core.mojo

```diff
@@ -1391,15 +1424,67 @@ fn get_mma_shape[
     else:
```

Comment:
What I'd recommend doing is to insert an `elif _is_amd_rdna():` at this level and move all the RDNA logic into that. You could lift the `elif _is_amd_gcn():` to this level, too, if you really felt that we should keep in the GCN checks, but I think we can safely drop those for now. It'll be a while before we're building out the GCN alternatives to the WMMA paths.

If you do that, you could leave the existing `else` path for CDNA entirely unchanged here, minimizing code changes and reducing the nesting of some of these if statements.